src/shared/components/app/navbar.tsx

```diff
@@ -44,6 +44,12 @@ export class Navbar extends Component<NavbarProps, NavbarState> {
     this.handleOutsideMenuClick = this.handleOutsideMenuClick.bind(this);
   }
 
+  componentWillMount(): void {
+    InboxService.observable.subscribe(() => {
+      this.render();
```

Comment:
Trying to get things to re-render here. It's not working still.

Reply:
Its more that on the initial `componentDidMount` , it would need to subscribe to each of those observables, and this would need a state to observe them. 

Observers are just one way of sharing state between non-parent-child components, some others are mentioned here that might be cleaner: https://stackoverflow.com/questions/38901106/how-to-make-a-shared-state-between-two-react-components